design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,430 @@
+    + [Lifecycle](#lifecycle)
+      - [Milestone 1](#milestone-1)
+        * [Installation](#installation)
+        * [Lifecycle](#lifecycle-1)
+        * [Update](#update)
+        * [Delete](#delete)
+    + [Order of Deployment](#order-of-deployment)
+      - [Milestone 1](#milestone-1-1)
+    + [Backwards compatibility](#backwards-compatibility)
+      - [Milestone 1](#milestone-1-2)
+    + [Performance](#performance)
+      - [Milestone 1](#milestone-1-3)
+    + [Affected Components](#affected-components)
+  * [Security](#security)
+  * [Test Plan](#test-plan)
+    + [Integration tests](#integration-tests)
+    + [Performance tests](#performance-tests)
+  * [Logs](#logs)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Glossary
+
+| **Term**                                                     | **Description**                                              |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| [Job](*https://kubernetes.io/docs/concepts/workloads/controllers/job/*) | K8s entity that runs a pod for one-time operation. Once the pod exists, the Job terminates |
+| [Deployment](*https://kubernetes.io/docs/concepts/workloads/controllers/deployment/*) | K8s entity that ensures multiple replications of a pod are running all the time. If a pod terminates, another one will be created. |
+| [Kubernetes Helm](https://helm.sh/)                          | A tool that streamlines and organizes the management of Kubernetes installation, deployment, and upgrade processes. |
+| [Helm Chart](https://helm.sh/docs/topics/charts/)            | A collection of files that describe a related set of Kubernetes resources. A chart can be used to describe a simple pod of a complex application. |
+
+## Useful links
+
+| Name                | Link                                                         |
+| ------------------- | ------------------------------------------------------------ |
+| Phase 2 Feature Doc | https://github.com/cyberark/secrets-provider-for-k8s/issues/102 |
+
+## Background
+
+The Cyberark Secrets Provider for K8s enables customers to use secrets stored and managed in Conjur/DAP vault and consume them as K8s Secrets for their application containers.
+
+During Phase 1, an *init container* was deployed in the same Pod as the customer's applications to populate the K8s secrets defined in the pod application's manifest.
+
+### Motivation
+
+The current implementation of the Secrets Provider only supports one application and makes the deployment of the Secrets Provider deeply coupled to the customer's applications. 
+
+## Requirements
+
+The solution should support the following capabilities:
+
+###  Milestone 1 *(current)*
+
+- Secrets Provider runs as a separate entity, serving multiple application containers that run on multiple pods
```

Comment:
@InbalZilberman are you suggesting that the secrets-provider will serve more than one namespace and have only the service-account in its application identity for Conjur authentication?

@oburstein-hub is it possible to have a service-account that has access to secrets in more than one namespace?

Reply:
it is possible, but them you need to use ClusterRole / ClusterRoleBinding instead of Role / RoleBinding

Reply:
thanks.

Then @InbalZilberman it seems that it is technically possible but it will require changes in Conjur as today `namespace` is required for authentication. we can change it so that the requirement will be `namespace` *OR* `service-account`.

Coming to think about it, it is possible today because the `namespace` that we check is the namespace in which the secrets-provider pod will be deployed in. So the application identity will still have the `namespace` attribute even if the service account serves multiple namespaces.

Reply:
> are you suggesting that the secrets-provider will serve more than one namespace and have only the service-account in its application identity for Conjur authentication?

@InbalZilberman please clarify (: